lib/storage-service.js

```diff
@@ -65,19 +65,29 @@ class StorageService {
    * @returns {Promise}
    */
   lookup(token, secret, ProviderType) {
-    return this._getLeaseManager(token, secret, ProviderType).then((manager) => {
+    let manager = null;
+
+    return this._getLeaseManager(token, secret, ProviderType).then((m) => {
+      manager = m;
+
       if (manager.status === STATUS.READY) {
+        StorageService.logManagerEvent('INFO', 'Manager was able to lookup a secret.', manager);
+
         // LeaseManager is ready, immediately resolve with data
         return manager.data;
       }
 
       // Queue the promise states with a timeout
-      return onceWithTimeout(manager, 'ready', this.timeout)
-        .then(() => manager.data)
-        .catch((err) => {
-          this._managers.delete(manager.name);
-          throw err;
-        });
+      return onceWithTimeout(manager, 'ready', this.timeout).then(() => {
+        StorageService.logManagerEvent('INFO', 'Manager was able to lookup a secret.', manager);
+
+        return manager.data;
+      })
+      .catch((err) => {
+        StorageService.logManagerEvent('ERROR', err, manager);
+        this._managers.delete(manager.name);
```

Comment:
Do you want to check for `manager === null` here in case `this._getLeaseManager` throws?